src/components/generic/Popup/Popup.jsx

```diff
@@ -0,0 +1,67 @@
+import React, {useEffect, useRef, useState} from 'react';
+import PropTypes from 'prop-types';
+import webexComponentClasses from '../../helpers';
+import {useElementPosition} from '../../hooks';
+
+/**
+ * Popup component - wrapper component that decides based on the available space
+ * where to be rendered - above or under the target element
+ *
+ * @param {object} props Data passed to the component
+ * @param {string} props.className  Custom CSS class to apply
+ * @param {object} props.style  Custom style to apply
+ * @param {React.ReactNode[]} props.children  List of children
+ * @returns {object}  JSX of the element
+ */
+export default function Popup({className, style, children}) {
+  const [cssClasses] = webexComponentClasses('popup', className);
+  const [customStyle, setCustomStyle] = useState(null);
+  const ref = useRef();
+
+  const parentPosition = useElementPosition(ref.current?.parentNode);
+  const childPosition = useElementPosition(ref.current?.firstChild);
+
+  useEffect(() => {
+    if (parentPosition && childPosition) {
+      const availableSpaceBelow = window.innerHeight - parentPosition.top - window.scrollY - 24;
+
+      if (
+        childPosition.height <= availableSpaceBelow ||
+        availableSpaceBelow >= parentPosition.top
+      ) {
+        setCustomStyle({
+          top: parentPosition.bottom + 4,
+          maxHeight: availableSpaceBelow - 24,
+          width: parentPosition.width,
+        });
+      } else {
+        setCustomStyle({
+          bottom: window.innerHeight - parentPosition.top - window.scrollY + 4,
+          maxHeight: parentPosition.top - 12,
+          width: parentPosition.width,
+        });
+      }
+    }
+
+    if (!customStyle) {
+      setCustomStyle({});
```

Comment:
immediately after the setCustomStyle calls above, the customStyle is not updated yet, so this will reset the custom style.

let's make use of the ifs above to prevent checking this not-yet-updated customStyle variable. And, for consistency with childPosition and parentPosition, let's call this `newPosition`:
```js
let newPosition;

if (!parentPosition || !childPosition) {
  position = {};
} else if (...) {
  position = {
    top: ...
  }
} else {
  position = {
     bottom: ...
  }
}

setPosition(newPosition);